datumaro/plugins/open_images_format.py

```diff
@@ -207,10 +217,144 @@ class OpenImagesImporter(Importer):
+                    'DisplayName': category.name,
+                })
+
+    def _save_label_category_parents(self):
+        all_label_names = set()
+        hierarchy_nodes = {}
+        orphan_nodes = []
+
+        def get_node(name):
+            return hierarchy_nodes.setdefault(name, {'LabelName': name})
+
+        for category in self._extractor.categories()[AnnotationType.label]:
+            all_label_names.add(category.name)
+
+            child_node = get_node(category.name)
+
+            if category.parent:
+                parent_node = get_node(category.parent)
+                parent_node.setdefault('Subcategory', []).append(child_node)
+            else:
+                orphan_nodes.append(child_node)
+
+        # The hierarchy has to be rooted in a single node. However, there's
+        # no guarantee that there exists only one orphan (label without a parent).
+        # Therefore, we create a fake root node and make it the parent of every
+        # orphan label.
+        # This is not a violation of the format, because the original OID does
+        # the same thing.
+        root_node = {
+            # Create an OID-like label name that isn't already used by a real label
+            'LabelName': next(root_name
+                for i in itertools.count()
+                for root_name in [f'/m/{i}']
+                if root_name not in all_label_names
+            ),
+            # If an orphan has no children, then it makes no semantic difference
+            # whether it's listed in the hierarchy file or not. So strip such nodes
+            # to avoid recording meaningless data.
+            'Subcategory': [node for node in orphan_nodes if 'Subcategory' in node],
+        }
+
+        hierarchy_path = osp.join(
+            self._save_dir, OpenImagesPath.ANNOTATIONS_DIR, OpenImagesPath.HIERARCHY_FILE_NAME)
+
+        with open(hierarchy_path, 'w', encoding='utf-8') as hierarchy_file:
+            json.dump(root_node, hierarchy_file, indent=4, ensure_ascii=False)
+            hierarchy_file.write('\n')
+
+    def _save_subsets(self):
+        label_categories = self._extractor.categories().get(
+            AnnotationType.label, LabelCategories())
+
+        for subset_name, subset in self._extractor.subsets().items():
+            if _RE_INVALID_SUBSET.fullmatch(subset_name):
+                raise UnsupportedSubsetNameError(item_id=next(iter(subset)).id, subset=subset)
+
+            image_description_name = f'{subset_name}-images-with-rotation.csv'
+            image_description_fields = [
+                'ImageID',
+                'Subset',
+                'OriginalURL',
+                'OriginalLandingURL',
+                'License',
+                'AuthorProfileURL',
+                'Author',
+                'Title',
+                'OriginalSize',
+                'OriginalMD5',
+                'Thumbnail300KURL',
+                'Rotation',
+            ]
```

Comment:
I think it's better to add this variable to the `OpenImagesPath` class, it will improve the readability of this function and allow the header to be used in different places (if it will be necessary). Same suggestion for `label_description_fields` variable.

Reply:
OK; done.